upi/vsphere/modules/aws/variables.tf

```diff
@@ -0,0 +1,95 @@
+variable "aws_config_version" {
+  description = <<EOF
+(internal) This declares the version of the AWS configuration variables.
+It has no impact on generated assets but declares the version contract of the configuration.
+EOF
+
+  default = "1.0"
+}
+
+variable "aws_extra_tags" {
+  type = "map"
+
+  description = <<EOF
+(optional) Extra AWS tags to be applied to created resources.
+
+Example: `{ "key" = "value", "foo" = "bar" }`
+EOF
+
+  default = {}
+}
+
+variable "aws_region" {
+  type        = "string"
+  description = "The target AWS region for the cluster."
+}
+
+variable "aws_master_availability_zones" {
+  type        = "list"
+  description = "The availability zones in which to create the masters. The length of this list must match master_count."
+}
+
+variable "aws_worker_availability_zones" {
+  type        = "list"
+  description = "The availability zones to provision for workers.  Worker instances are created by the machine-API operator, but this variable controls their supporting infrastructure (subnets, routing, etc.)."
+}
+
+variable "bootstrap_ip_address" {
+  type = "string"
+}
+
+variable "bootstrap_count" {
+  type = "string"
+}
+
+variable "master_ip_addresses" {
+  type = "list"
+}
+
+variable "master_count" {
+  type = "string"
+}
+
+variable "worker_ip_addresses" {
+  type = "list"
+}
+
+variable "worker_count" {
+  type = "string"
+}
+
+variable "cluster_id" {
+  type        = "string"
+  description = "This cluster id must be of max length 27 and must have only alphanumeric or hyphen characters."
+}
+
+variable "base_domain" {
+  type        = "string"
+  description = "The base DNS zone to add the sub zone to."
+}
+
+variable "cluster_domain" {
+  type        = "string"
+  description = "The base DNS zone to add the sub zone to."
+}
+
+variable "vpc_id" {
+  type = "string"
+}
+
+variable "machine_cidr" {
+  type = "string"
+}
+
+variable "aws_public_subnet_id" {
+  type = "list"
+}
+
+variable aws_private_subnet_id {
+  type = "list"
+}
+
+// TODO: this might need to be a list...but may not work right with vsphere on aws
```

Comment:
Do you want to address this before merging?  I don't see a problem with a single zone for Terraform modules we use for internal testing.  Folks who want a vSphere cluster are probably not going to be using AWS, because if you want a cluster on AWS you'd just use our native AWS providers, right?